agents.py

```diff
@@ -569,10 +571,11 @@ def random_location_inbounds(self, exclude=None):
     def delete_thing(self, thing):
         """Deletes thing, and everything it is holding (if thing is an agent)"""
         if isinstance(thing, Agent):
-            for obj in thing.holding:
-                super().delete_thing(obj)
-                for obs in self.observers:
-                    obs.thing_deleted(obj)
+            # for obj in thing.holding:
+            #     super().delete_thing(obj)
+            #     for obs in self.observers:
+            #         obs.thing_deleted(obj)
```

Comment:
I don't see a reason to comment these lines out. Even though the second line is redundant, the observers of the environment still need to be informed about each thing being deleted.

Reply:
if we delete all of the agent holdings with `del thing.holding`, and thanks to python memory management. All instants of the agent's holding would vanish from memory even in the observer list. Tell me what you think.

Reply:
The observers of the environment need to be updated about all the things getting deleted along with the agent. I have not seen `observers` used anywhere, so it depends upon the maintainers of the project to keep or exclude it!